lisp/ess-r-flymake.el

```diff
@@ -0,0 +1,149 @@
+            (unwind-protect
+                ;; Only proceed if `proc' is the same as
+                ;; `ess-r--flymake-proc', which indicates that
+                ;; `proc' is not an obsolete process.
+                (if (with-current-buffer source (eq proc ess-r--flymake-proc))
+                    (with-current-buffer (process-buffer proc)
+                      (goto-char (point-min))
+                      ;; Parse the output buffer for diagnostic's
+                      ;; messages and locations, collect them in a list
+                      ;; of objects, and call `report-fn'.
+                      (cl-loop
+                       while (search-forward-regexp
+                              ;; Regex to match the output lint() gives us.
+                              (rx line-start "<text>:"
+                                  ;; row
+                                  (group-n 1 (one-or-more num)) ":"
+                                  ;; column
+                                  (group-n 2 (one-or-more num)) ": "
+                                  ;; type
+                                  (group-n 3 (| "style: " "warning: " "error: "))
+                                  ;; msg
+                                  (group-n 4 (one-or-more not-newline)) "\n")
+                              nil t)
+                       for msg = (match-string 4)
+                       for (beg . end) = (flymake-diag-region
+                                          source
+                                          (string-to-number (match-string 1))
+                                          (string-to-number (match-string 2)))
+                       for type = (let ((str (match-string 3)))
+                                    (cond ((string-equal str "error: ") :error)
+                                          ((string-equal str "warning: ") :warning)
+                                          ((string-equal str "style: ") :note)))
+                       collect (flymake-make-diagnostic source
+                                                        beg
+                                                        end
+                                                        type
+                                                        msg)
+                       into diags
+                       finally (funcall report-fn diags)))
+                  (flymake-log :warning "Canceling obsolete check %s"
+                               proc))
+              ;; Cleanup the temporary buffer used to hold the
+              ;; check's output.
+              (kill-buffer (process-buffer proc))))))))))
+
+(defun ess-r-setup-flymake ()
+  "Setup flymake for ESS."
+  (when (< 26 emacs-major-version)
+    (error "ESS-flymake requires Emacs version 26 or later"))
+  (add-hook 'flymake-diagnostic-functions #'ess-r-flymake nil t)
+  ;; Try not to enable flymake if flycheck is already running:
+  (unless (bound-and-true-p flycheck-mode)
```

Comment:
A message might be useful here to inform people why flymake is not activated.

Reply:
Done